pkg/reconciliation/reconcile_racks.go

```diff
@@ -2243,6 +2246,52 @@ func (rc *ReconciliationContext) fixMissingPVC() (bool, error) {
 	return false, nil
 }
 
+func (rc *ReconciliationContext) SetFullQueryLogging() result.ReconcileResult {
+	datacenter := rc.GetDatacenter()
+	var DCConfig map[string]interface{}
+	if err := json.Unmarshal(datacenter.Spec.Config, &DCConfig); err != nil {
+		rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+		return result.RequeueSoon(2)
+	}
+	shouldFQLBeEnabled := false
+	if err := DCConfig["full_query_logging_options"]; err == nil {
```

Comment:
What if err is not nil?

Reply:
Also, instead of checking for error, the correct use for map is:

```go
if fqlValue, found := dcConfig["full_query_logging_options"]; found { }
```

Reply:
So if `err != nil` then `shouldFQLBeEnabled = true` is never run which makes `shouldFQLBeEnabled` always false. It will end up with `rc.NodeMgmtClient.CallSetFullQuerylog(podPtr, false)` being run, which will disable FQL as intended.

> if fqlValue, found := dcConfig["full_query_logging_options"]; found { }

You're right, my original code actually makes very limited sense, this variable definitely should not have been called `err`! Thanks for picking this up.

If you have suggestions to add further clarity on the conditional logic I'm open to them, I don't particularly like the way this is written currently but it seemed to be the way to do it in Go.